File: 8 kyu/Total pressure calculation/solution.ts
export function doubleton(num: number): number {
  num += 1
  while(new Set(""+num).size != 2){
    num++
  }
  return num
} 
__________________
export function doubleton(num: number) {
   while(num++){
        const numbers = String(num).split('')
        if(new Set(numbers).size === 2) {
            return +numbers.join('')
        }
   }
}
________________
export function doubleton(n: number): number {
  for(let i = n + 1; i < 1_000_000; i++){
    if((new Set(i.toString())).size==2)
        return i;
  }
  return 1
} 
___________________
export function doubleton(num: number): number {
  for (let n = num + 1; ; n++)
    if (new Set([...`${n}`]).size === 2)
      return n;
} 
_________________
export function doubleton(num: number): number {
  let num2 = (num + 1).toString()
  while ([...new Set(num2.split(''))].join('').length !== 2) num2 = ((+num2) + 1).toString()
  return +num2
} 

File: 8 kyu/Total pressure calculation/solution.rs
use std::collections::HashSet;

fn doubleton(num: u32) -> u32 {
    let mut n=num+1;
    loop {
        if n.to_string().chars().collect::<HashSet<char>>().len() == 2 {
            break;
        }
        n+=1;
    }
    n
}

#############
fn doubleton(num: u32) -> u32 {
    ((num + 1)..)
        .find(|&i| digits(i).fold(0u32, |m, x| m | 1 << x % 10).count_ones() == 2)
        .unwrap()
}

fn digits(x: u32) -> impl Iterator<Item = u32> {
    std::iter::successors(Some(x), |&x| if x >= 10 { Some(x / 10) } else { None })
}
  
###########
use std::collections::HashSet;

fn doubleton(num: u32) -> u32 {
    let mut result = num + 1;
    while !is_doubleton(result) {
        result += 1;
    }
    result
}

fn is_doubleton(num: u32) -> bool {
    num.to_string().chars().collect::<HashSet<_>>().len() == 2
}
  
#################
fn doubleton(num: u32) -> u32 {
    let mut n = num;
    if is_doubleton(n) {n += 1;}
    while !is_doubleton(n) {n += 1;}
    n
}

fn is_doubleton(num: u32) -> bool {
    let mut s = num.to_string().into_bytes();
    s.sort();
    s.dedup();
    s.len() == 2
}
  
############
fn doubleton(mut n: u32) -> u32 {
    n += 1;
    let mut a = n; let mut b = None; let mut c = None;
    while a != 0 {
        let d = a % 10;
        if b.is_none() { b = Some(d); } else if c.is_none() {
            if b != Some(d) {c = Some(d); }
        } else if b != Some(d) && c != Some(d) {
            n += 1;a = n;b = None;c = None;continue;
        }
        a /= 10;
        if a==0 && c.is_none(){n += 1;a = n;b = None;}
    }
    n
}
